### ISSUE TITLE
Question: Reducers in non-route containers that are included in multiple routes/containers

### ISSUE THREAD

**Issue by SimonSelg** [react-boilerplate/react-boilerplate#709]
Hey,

the docs state:

![image](https://cloud.githubusercontent.com/assets/9430680/16800581/461daf70-48f6-11e6-8b75-ddf1a5bb6ce9.png)

However, that only works when the container is only included in _one_ parent, because if it's included in multiple parents _reducer composition_ still works, but the state the reducer has access to will be a different substate of the global state, depending on the parent forms substate. 

Example: We have two containers "NewContactPage" and "ContactDetailPage", which correspond the the "newContactPage" and "contactDetailPage" substate of the global state.

Now we have another container, "ContactDetailForm", which is a child of both "NewContactPage" and "ContactDetailPage". When applying reducer composition, as stated in the docs, the reducer of "ContactDetailForm" has access the "newContactPage" or the "contactDetailPage", depending on the route. Now a selector in ContactDetailForm's selectors has no way of selecting any value from that state, because it doesn't know if the data is in "state.contactDetailPage" or "state.newContactPage".

Does anyone have a quick solution for this besides including the reducer in reducers.js? 

The only solution I could think of was to pass the substate name as a prop to "ContactDetailForm" (as reselect selectors have access to the props), and then selecting the right substate depending on that prop. But that isn't really portable in my opinion....

Cheers!

**Comment by mxstbr** [comment]
To be honest I've never had that usecase so far, so I've never done it. Could you add a wrapping container (i.e. `wrapper -> (NewContactPage & ContactDetailPage)` that handles the choosing?

**Comment by timothyallan** [comment]
I'm in the same boat. I want to make a semi-global UI reducer and am figuring out how to best implement it.

Another use case: I've got various components are able to open/close various dialog boxes at any point. I need to maintain the visibility state of those dialogs so that Component A can know that Dialog A is already open, and won't open another, even if Dialog A was opened by Component B. 

Both components need access to the same UI state chunk, but the components are totally separate.

**Comment by mxstbr** [comment]
> Both components need access to the same UI state chunk, but the components are totally separate.

Why not make the dialog a container with a reducer that maintains it's own state?

**Comment by timothyallan** [comment]
That's doing my head in thinking about it.

Lets say there are two different dialog boxes, A and B, and either can be popped open/closed via the MenuLinks, or via the WizardContent components (or any other number of components, but we'll just use 2 for now).

```
<Page>
  <Header>
    <MenuLinks/>
  </Header>
  <WizardContent/>
</Page>
```

Making the dialogs into containers with their own reducers would mean that I'd always have to have them just hanging around in the main root of the app, no? 

```
<Page>
  <Header>
    <MenuLinks/>
  </Header>
  <WizardContent/>
  <DialogA/>
  <DialogB/>
</Page>
```

Then I'd just have to pass their selectors into any component that needed to interact with them?

**Comment by mxstbr** [comment]
Yeah, I think that's the best way? Then you can do

```JSX
// MenuLinks/index.js

import { openDialogA } from 'containers/DialogA/actions';

/* … */

<button onClick={openDialogA} />
```

**Comment by mvid** [comment]
We are having a similar issue. 

We have a data entity that isn't contained to any one Container. We created a top level store key for it, and created a Container just for managing the reducer, saga, actions, constants for the data entity. Our data processor Container will not be connected to a Route, because it will never display anything, and would never be accessed in such a way.

We have hooked up the reducer in the external `reducer.js` file, but we can't figure out a way to make the sagas listen for actions...
